contracts/DrawAndPrizeDistributionTimelock.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.6;
+import "@pooltogether/v4-core/contracts/interfaces/IDrawBeacon.sol";
+import "@pooltogether/v4-core/contracts/interfaces/IDrawBuffer.sol";
+import "@pooltogether/owner-manager-contracts/contracts/Manageable.sol";
+import "./interfaces/IDrawCalculatorTimelock.sol";
+
+interface IPrizeDistributionFactory {
+  function pushPrizeDistribution(uint32 _drawId, uint256 _totalNetworkTicketSupply) external;
+}
+
+/**
+  * @title  PoolTogether V4 IPrizeDistributionTimelock 
+  * @author PoolTogether Inc Team
+  * @notice The IPrizeDistributionTimelock smart contract interface...
+*/
+interface IDrawAndPrizeDistributionTimelock  {
+  function push(IDrawBeacon.Draw memory draw, uint256 totalNetworkTicketSupply) external;
+}
+
+/**
+  * @title  PoolTogether V4 PrizeDistributionTimelock
+  * @author PoolTogether Inc Team
+  * @notice The PrizeDistributionTimelock smart contract...
+*/
+contract DrawAndPrizeDistributionTimelock is IDrawAndPrizeDistributionTimelock, Manageable {
+  
+  /* ============ Global Variables ============ */
+
+    /// @notice The DrawBuffer contract address.
+    IDrawBuffer public immutable drawBuffer;
+
+    /// @notice Internal PrizeDistributionFactory reference.
+    IPrizeDistributionFactory public immutable prizeDistributionFactory;
+
+    /// @notice Timelock struct reference.
+    IDrawCalculatorTimelock public timelock;
+
+  /* ============ Events ============ */
+    /// @notice Emitted when the contract is deployed.
+    event Deployed(
+        IDrawBuffer indexed drawBuffer,
+        IPrizeDistributionFactory indexed prizeDistributionFactory,
+        IDrawCalculatorTimelock indexed timelock
+    );
+
+    /**
+     * @notice Emitted when Draw and PrizeDistribution are pushed to external contracts.
+     * @param drawId Draw ID
+     * @param draw Draw
+     * @param totalNetworkTicketSupply totalNetworkTicketSupply
+     */
+    event DrawAndPrizeDistributionPushed(
+      uint32 indexed drawId, 
+      IDrawBeacon.Draw draw, 
+      uint256 totalNetworkTicketSupply
+    );
+
+
+  /* ============ Constructor ============ */
+
+    /**
+     * @notice Initialize DrawAndPrizeDistributionTimelock smart contract.
+     * @param _owner The smart contract owner
+     */
+    constructor(
+      address _owner,
+      IDrawBuffer _drawBuffer,
+      IPrizeDistributionFactory _prizeDistributionFactory,
+      IDrawCalculatorTimelock _timelock) Ownable(_owner) 
+    {
+      drawBuffer = _drawBuffer;
+      prizeDistributionFactory = _prizeDistributionFactory;
+      timelock = _timelock;
+
+      emit Deployed(_drawBuffer, _prizeDistributionFactory, _timelock);
+    }
+
+  function push(IDrawBeacon.Draw memory _draw, uint256 _totalNetworkTicketSupply) external override onlyManagerOrOwner {
+      timelock.lock(_draw.drawId, _draw.timestamp + _draw.beaconPeriodSeconds);
```

Comment:
We call `_draw.drawId` 3 times so we can store it in a variable.

Reply:
Why?